### PR TITLE
Add ruff for linting, remove flake8, remove isort, remove pylint

### LINKED ISSUE #88: Switch to using ruff for Python linting and code formatting
### Problem Description
- We currently use `isort`, `flake8`, and `pydocstyle` for managing out python linting and code formatting.
- However, many libraries have switched to using `ruff`. It is a much faster linter/formatter and can replace the `isort`/`flake8`/`pydocstyle` libraries (among others)

### Expected behavior
- Use `ruff` as the standard linter/formatter

### PR DESCRIPTION
**Pull request by gsheni**
- Added [ruff](https://github.com/charliermarsh/ruff) for running linting
- Added **ruff** commands to Makefile
- Removed **flake8** as dev dependency, since **ruff** does this, but faster
- Remove **isort** as dev dependency, since **ruff** has the [same function](https://github.com/charliermarsh/ruff#how-does-ruffs-import-sorting-compare-to-isort), but faster
- Removed **pylint** as dev dependency, since **ruff** has [more implemented rules](https://docs.astral.sh/ruff/faq/#how-does-ruffs-linter-compare-to-pylint)
- Removed setup.cfg
- Closes #88

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -81,9 +81,6 @@ install-develop: clean-build clean-pyc ## install the package in editable mode a
 
 .PHONY: lint-deepecho
 lint-deepecho: ## check style with flake8 and isort
```
Comment (amontanez24) [inline Makefile:83]: do we need this make command if it doesn't do anything?

File: deepecho/sequences.py [hunk 2]
```diff
@@ -110,27 +112,29 @@ def _convert_to_dicts(segments, context_columns):
         if context_columns:
             context = segment[context_columns]
             if len(context.drop_duplicates()) > 1:
-                raise ValueError('Context columns are not constant within each segment.')
```
Comment (amontanez24) [inline deepecho/sequences.py:113]: Is there any way to allow single lines like this?

File: Makefile (hunk 1 above)
Comment (gsheni) [reply Makefile]: Removed this and another unused lint command:
https://github.com/sdv-dev/DeepEcho/pull/94/commits/e4957e79af60a2902c2fce59d9035cc11af933b4

File: deepecho/sequences.py (hunk 2 above)
Comment (gsheni) [reply deepecho/sequences.py]: Sure, increased line-length to 99
https://github.com/sdv-dev/DeepEcho/pull/94/commits/42f595f77ae29cc0ef7ccecd96a2adf48bac5ee9

**Review by amontanez24 — APPROVED** [review]

**Pull request merged**